elide-core/src/main/java/com/yahoo/elide/core/sort/Sorting.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core.sort;
+
+import com.yahoo.elide.core.EntityDictionary;
+import lombok.ToString;
+
+import javax.ws.rs.core.MultivaluedMap;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Generates a simple wrapper around the sort fields from the JSON-API GET Query.
+ */
+@ToString
+public class Sorting {
+
+    /**
+     * Denotes the intended sort type from json-api field.
+     */
+    public enum SortOrder { asc, desc }
+
+    private static final Sorting DEFAULT_EMPTY_INSTANCE = new Sorting(null);
+
+    private final Map<String, SortOrder> sortRules = new LinkedHashMap<>();
+    private volatile Map<Class<?>, Map<String, SortOrder>> validSortRules = null;
+
+
+    /**
+     * Constructs a new Sorting instance
+     * @param sortingRules The map of sorting rules
+     */
+    public Sorting(final Map<String, SortOrder> sortingRules) {
+        if (sortingRules != null && !sortingRules.isEmpty()) {
+            sortRules.putAll(sortingRules);
+        }
+    }
+
+    /**
+     * Checks to see if the sorting rules are valid for the given JPA class
+     * @param entityClass The target jpa entity
+     * @param dictionary The elide entity dictionary
+     * @param <T> The Type of the target entity
+     * @return The validity of the sorting rules on the target class
+     */
+    public <T> boolean hasValidSortingRules(final Class<T> entityClass,
+                                        final EntityDictionary dictionary) {
+        if (validSortRules == null) {
+            validSortRules = new LinkedHashMap<>();
+        }
+        final Map<String, SortOrder> validSortingRules = new LinkedHashMap<>();
+        final List<String> entities = dictionary.getAttributes(entityClass);
+        if (entities != null) {
+            entities.stream()
+                    .filter(sortRules::containsKey)
```

Comment:
When we parse the query params we don't know the target entity type yet. In the filters, it has a filter[entity.field] so you know the target entity, however since we are only sorting the terminal resource or collection, we won't know the actual terminal class type until we hit one of the three TerminalState's.